File: retos_semana_1/Nutricion/reto_1_nutricion.py
# Reto 1 Nutricion
# Realizado Por: Jose Villero ♠

def solucion(edad, peso):
    edad = int(input('Indicar la edad del paciente: '))
    peso = float(input('Indicar el peso del paciente en kilogramos: '))

    APORTE_CH = 60.1         # Aporte Carbohidrato
    APORTE_P = 30.5          # Aporte proteinas
    APORTE_F_V = -24.4       # Aporte frutas y verduras

    if 5 <= edad <= 10:     # Verificacion de edades entre 5 a 10 años
        if peso < 16:
            desnutricion = (APORTE_CH * 2) + (APORTE_P * 1) + (APORTE_F_V * 2)
            estado = 'A'
            dias = round(((22 - peso) / (desnutricion / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        elif peso > 28:
            sobrepeso = (APORTE_CH * 0.6) + (APORTE_P * 1) + (APORTE_F_V * 4)
            estado = 'B'
            dias = round(((peso - 24) / (sobrepeso / (-1000))) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        else:
            pesoSaludable = (APORTE_CH * 0.5) + (APORTE_P * 0.7) + (APORTE_F_V * 2)
            estado = 'C'
            dias = round(((28 - peso) / (pesoSaludable / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance el peso maximo')
            
    elif 10 < edad <= 13:           # Verificacion de edades entre 10 a 13 años
        if peso < 30:
            desnutricion = (APORTE_CH * 2) + (APORTE_P * 1) + (APORTE_F_V * 2)
            estado = 'A'
            dias = round(((32 - peso) / (desnutricion / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        elif peso > 50:
            sobrepeso = (APORTE_CH * 0.6) + (APORTE_P * 1) + (APORTE_F_V * 4)
            estado = 'B'
            dias = round(((peso - 24) / (sobrepeso / (-1000))) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        else:
            pesoSaludable = (APORTE_CH * 0.5) + (APORTE_P * 0.7) + (APORTE_F_V * 2)
            estado = 'C'
            dias = round(((50 - peso) / (pesoSaludable / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance el peso maximo')
            
    elif 13 < edad <= 17:           # Verificacion de edades entre 13 a 17 años
        if peso < 51:
            desnutricion = (APORTE_CH * 2) + (APORTE_P * 1) + (APORTE_F_V * 2)
            estado = 'A'
            dias = round(((56 - peso) / (desnutricion / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        elif peso > 63:
            sobrepeso = (APORTE_CH * 0.6) + (APORTE_P * 1) + (APORTE_F_V * 4)
            estado = 'B'
            dias = round(((peso - 24) / (sobrepeso / (-1000))) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance un peso saludable')
        else:
            pesoSaludable = (APORTE_CH * 0.5) + (APORTE_P * 0.7) + (APORTE_F_V * 2)
            estado = 'C'
            dias = round(((63 - peso) / (pesoSaludable / 1000)) + 0.5)
            print(
                f'El estado nutricional del paciente es {estado} y se requieren {dias} dias de dieta para que alcance el peso maximo')
            
    else:          # Si la edad es mayor de 17 y menor de 15 muestra mensaje de error      
        if edad > 17:
            print('Error: La edad del paciente es mayor a 17 años')
        elif edad < 5:
            print('Error: La edad del paciente es menor de 5 años')

File: retos_semana_1/Nutricion/Reto1nutricion.java
// Realizado por Jose Villero
// Reto 1 - Nutricion

import java.util.Scanner;
import java.lang.Math;

public class Reto1nutricion {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int edad;
        double peso, pesoSaludable, sobrepeso, desnutricion, dias;
        char estado;
        double aporteCH = 60.1; // aporte calorico de carbohidratos
        double aporteP = 30.5;  // aporte calorico de proteinas
        double aporteFV = -24.4;    // aporte calorico de frutas y verduras

        // Solicitar datos de edad y peso
        System.out.println("Ingrese su edad: ");
        edad = sc.nextInt();
        System.out.println("Ingrese su peso: ");
        peso = sc.nextDouble();

        // Verificacion de edades entre 5 a 10 años
        if (edad >= 5 && edad <= 10){
            if (peso < 16){
                desnutricion = (aporteCH * 2) + (aporteP * 1) + (aporteFV * 2);
                estado = 'A';
                dias = Math.round(((22 - peso) / (desnutricion / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            } else if (peso > 28) {
                sobrepeso = (aporteCH * 0.6) + (aporteP * 1) + (aporteFV * 4);
                estado = 'B';
                dias = Math.round(((peso - 24) / (sobrepeso / (-1000))) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            }else{
                pesoSaludable = (aporteCH * 0.5) + (aporteP * 0.7) + (aporteFV * 2);
                estado = 'C';
                dias = Math.round(((28 - peso) / (pesoSaludable / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance el peso maximo.");
            }
            // Verificacion de edades entre 10 y 13 años
        } else if (edad > 10 && edad <= 13) {
            if (peso < 30){
                desnutricion = (aporteCH * 2) + (aporteP * 1) + (aporteFV * 2);
                estado = 'A';
                dias = Math.round(((32 - peso) / (desnutricion / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            } else if (peso > 50) {
                sobrepeso = (aporteCH * 0.6) + (aporteP * 1) + (aporteFV * 4);
                estado = 'B';
                dias = Math.round(((peso - 24) / (sobrepeso / (-1000))) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            } else {
                pesoSaludable = (aporteCH * 0.5) + (aporteP * 0.7) + (aporteFV * 2);
                estado = 'C';
                dias = Math.round(((50 - peso) / (pesoSaludable / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance el peso maximo.");
            }
            // Verificacion de edades entre 13 a 17 años
        } else if (edad > 13 && edad <= 17) {
            if (peso < 51){
                desnutricion = (aporteCH * 2) + (aporteP * 1) + (aporteFV * 2);
                estado = 'A';
                dias = Math.round(((56 - peso) / (desnutricion / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            } else if (peso > 63) {
                sobrepeso = (aporteCH * 0.6) + (aporteP * 1) + (aporteFV * 4);
                estado = 'B';
                dias = Math.round(((peso - 24) / (sobrepeso / (-1000))) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance un peso saludable.");
            }else {
                pesoSaludable = (aporteCH * 0.5) + (aporteP * 0.7) + (aporteFV * 2);
                estado = 'C';
                dias = Math.round(((63 - peso) / (pesoSaludable / 1000)) + 0.5);
                System.out.println("El estado nutricional del paciente es " + estado + " y se requieren " + dias + " dias de dieta para que alcance el peso maximo.");
            }
        } else {
            if (edad > 17){
                System.out.println("Error: La edad del paciente es mayor a 17 años");
            } else if (edad < 5) {
                System.out.println("'Error: La edad del paciente es menor de 5 años'");
            }
        }
    }
}
